src/core.js

```diff
@@ -732,7 +732,11 @@ define([
 
         if (target) {
             // the .change fires the validation controls
-            target.val(target.val() + path).change();
+            if (this.opts().features.rich_text) {
+                target.ckeditor().editor.insertHtml(widgets.util.toRichText(path, mug.form, true));
```

Comment:
Could get the form here more reliably with `_this.data.core.form`. The problem here is that sometimes `mug` can be `null` or `undefined`, which would break this.